static/app/components/core/select/index.tsx

```diff
@@ -542,12 +531,13 @@ function SelectPicker<OptionType extends OptionTypeBase>({
     Component = ReactSelect;
   }
 
-  return <Component ref={forwardedRef} {...props} />;
+  return <Component ref={ref as any} {...props} />;
 }
 
-// The generics need to be filled here as forwardRef can't expose generics.
-export const Select = forwardRef<typeof ReactSelect<GeneralSelectValue>, ControlProps>(
-  function RefForwardedSelectControl(props, ref) {
-    return <SelectControl forwardedRef={ref as any} {...props} />;
-  }
-);
+// XXX (tkdodo): this type assertion is a leftover from when we had forwardRef
+// Omit on the ControlProps messes up the union type
+// the fix is to remove this type assertion, export Select directly and fix the type issues
+export const Select = SelectControl as (
+  props: Omit<ControlProps, 'ref'> &
+    React.RefAttributes<typeof ReactSelect<GeneralSelectValue>>
+) => React.JSX.Element;
```

Comment:
as the comment says, I’ve kept the previous types to avoid type error regressions. If we export `SelectControl` directly, we get 32 type errors in 24 files. I’m not even sure if this is worth fixing, I think we should rather move away from react-select in the mid-term.